### PR TITLE
Workaround CMake requirement for mender-client 4.0.x series

### PR DESCRIPTION
**Pull request by lluiscampos**

### CODE REVIEW

**Review by danielskinstad — APPROVED** [review]

File: mender-deb-package [hunk 1]
```diff
@@ -235,12 +235,20 @@ EOF
...
+          -a "$OS_CODENAME" = "buster" \
+          -a "$ARCH" = "armhf" ];
+  then
+    local REPO_PATH="${REPO_URL#https:/}"
+
+    # mender requires CMake 3.14 when using MENDER_DOWNLOAD_BOOST but buster-armhf (Raspberry Pi OS)
+    # image comes with CMake 3.13. We workaround the issue by pre-installing BOOST in the image,
+    # so here we can ignore the requirement.
+    if grep -q cmake_minimum_required "$REPO_PATH/CMakeLists.txt"; then
+      sed -i.bak '/cmake_minimum_required/d' $REPO_PATH/CMakeLists.txt
```
Comment (kacf) [inline mender-deb-package:249]: Maybe pre-existing, but will this trigger this behavior?

```
$ mender-update --version
4.0.4~dirty
```

File: mender-deb-package [hunk 2]
```diff
@@ -249,6 +249,14 @@ EOF
...
       diff -u $REPO_PATH/CMakeLists.txt.bak $REPO_PATH/CMakeLists.txt || true
     fi
+
+    # On 4.0.x series, the vendored projects are added which indirectly rises the CMake required
+    # version. These are not needed anyway so we can remove it. See:
+    # https://github.com/mendersoftware/mender/pull/1610/commits/c7c48d1b9368ce3d4be53a4e29536d324b328f82
+    if grep -q "add_subdirectory(vendor/expected)" "$REPO_PATH/main.cmake"; then
+      sed -i.bak '\|add_subdirectory(vendor/expected)|d' $REPO_PATH/main.cmake
+      diff -u $REPO_PATH/main.cmake.bak $REPO_PATH/main.cmake || true
+    fi
```
Comment (kacf) [inline mender-deb-package:259]: This commit can be dropped after fixing the client, right?

Comment (lluiscampos) [reply mender-deb-package:259]: If we want to rebuild existing tags (4.0.2) we still would need it.

Side note: I am working on something else that will allow me to remove the two hacks. I'll update this PR ~soon~ at some point.

File: mender-deb-package (hunk 1 above)
Comment (lluiscampos) [reply mender-deb-package:249]: You are probably right! I missed it when I reviewed Daniel's work.

**Pull request closed without merging**